### ISSUE TITLE
[firebase_messaging] onTokenRefresh is running twice

### ISSUE THREAD

**Issue by Inambe** [FirebaseExtended/flutterfire#2528]
**Describe the bug**
I declared `onTokenRefresh`'s listener in `initState` but it runs twice when app is installed and runs once when app is restarted (when actually token isn't refreshed/changed).

This is my `initState` method:
```dart
@override
	void initState() {
		super.initState();
		var fcmTokenStream = firebaseMessaging.onTokenRefresh;
		onTokenRefreshListen = fcmTokenStream.listen((token) async {
			print("calling saveToken @ onTokenRefresh listen");
			await TokenAPI.saveToken(
				token: token
			);
		});
	}
```

and `dispose` method:
```dart
@override
	void dispose() {
		super.dispose();
		onTokenRefreshListen.cancel();
	}
```

**To Reproduce**
1. add a listener to onTokenRefresh
2. it'll run with old token

**Expected behavior**
It should not run when app is started but when a token is refreshed/changed. If I even accept that it'll run on app startup still it should not run twice with the same token when app is reinstalled.

**Flutter doctor**
Run `flutter doctor` and paste the output below:

```
[✓] Flutter (Channel stable, v1.17.0, on Linux, locale en_US.UTF-8)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
[✓] Android Studio (version 3.6)
[✓] Connected device (1 available)

• No issues found!

```

**Comment by Inambe** [comment]
For now (cause this app is in production) I'm using:
```dart
var fcmTokenStream = firebaseMessaging.onTokenRefresh.distinct();
```
So if `onTokenRefresh` stream is even being changed two times (with the same token) listener is called only once.

**Comment by TahaTesser** [comment]
Issue does exist
closing as duplicate of https://github.com/FirebaseExtended/flutterfire/issues/1824
